### PR TITLE
Use list of points instead of vararg for specifying on the fly trajectories.

### PR DESCRIPTION
**Pull request by brennenputh**
This PR changes the generatePath method to accept a list of PathPoints instead of taking two arguments and a vararg of PathPoints.

The reasoning behind this change is situations like my own, where I would like to be able to pass several points into a function to get a command that runs a path.  Using the PathPlanner generatePath method everywhere would mean unnecessary code duplication of the constraints (I'm only running one robot, the constraints don't change during runtime).  This causes me to have to write ugly code in order to accept a list of points, getting the first two by index and then calling subList and converting that sub list to an array.

The secondary reasoning is that the paths are simply reconstructed into an ArrayList at the end of the day already.  It doesn't make sense to convert away from a List back into a List.

As other options on this PR, I'm unsure whether it would be better to keep a vararg overload.  If this is desired, I can add one.  Also, it may be easier to add overloads that take Lists, however this adds a lot of unnecessary code for the Lists to be turned into arrays for varargs to be turned right back into a List.

Feel free to deny if this is actively not desired.

### CODE REVIEW

**Comment by mjansen4857** [comment]
Instead of replacing the methods, I would prefer overloads. Using varargs makes it a bit cleaner to use in some cases so I'd like to keep that option.

For example:
```Java
generatePath(constraints, currentRobotPoint, loadingStationPoint);
```

vs

```Java
generatePath(constraints, List.of(currentRobotPoint, loadingStationPoint));
```

It just removes the need to create a list yourself. All the vararg overload would need to do is create the list for you.

**Comment by brennenputh** [comment]
Added in latest commit.  I do have one question, what is the point of having a overload with the max velocity and max acceleration broken out instead of passed in as PathConstraints?  It seems like it adds a lot of overloads for not much gain.

**Comment by brennenputh** [comment]
One other thing, would it be wise to implement the same change in C++?  I am inexperienced in C++ so I am unsure if there is a standard.

**Comment by mjansen4857** [comment]
C++ could benefit from doing the same thing, just vectors and initializer lists instead of lists and varargs. I could take care of that though if you'd like.

**Comment by brennenputh** [comment]
I'll get the C++ changes done, they don't look hard from a glance.  I'll also make sure to mark the separate vel/accel as deprecated.

**Comment by brennenputh** [comment]
Looked around, I'm not sure what to use as the equivalent of deprecated in C++.  Other than that though, I believe it's done.

**Comment by mjansen4857** [comment]
It looks like you can do deprecation in C++ like this:
```C++
[[deprecated("Use someOtherMethod() instead.")]]
void deprecatedMethod() {}
```

Haven't bothered to try to compile it with the roborio toolchain so if it fails to build then no big deal.

**Comment by brennenputh** [comment]
Done.

File: pathplannerlib/src/main/native/include/pathplanner/lib/PathPlanner.h [hunk 1]
```diff
@@ -72,6 +72,75 @@ class PathPlanner {
...
+	 * As you can't see the path in the GUI when using this method, make sure you have a good idea
+	 * of what works well and what doesn't before you use this method in competition. Points positioned in weird
+	 * configurations such as being too close together can lead to really janky paths.
+	 *
+	 * @param constraints The max velocity and max acceleration of the path
+	 * @param reversed Should the robot follow this path reversed
+	 * @param points Points in the path
+	 * @return The generated path
+	 */
+	static PathPlannerTrajectory generatePath(
```
Comment (mjansen4857) [inline pathplannerlib/src/main/native/include/pathplanner/lib/PathPlanner.h:100]: Missed a deprecation here

File: pathplannerlib/src/main/native/include/pathplanner/lib/PathPlanner.h [hunk 2]
```diff
@@ -72,6 +72,75 @@ class PathPlanner {
...
+			units::meters_per_second_t const maxVel,
+			units::meters_per_second_squared_t const maxAccel,
+			bool const reversed, std::vector<PathPoint> const points) {
+		return generatePath(PathConstraints { maxVel, maxAccel }, reversed,
+				points);
+	}
+	;
```
Comment (mjansen4857) [inline pathplannerlib/src/main/native/include/pathplanner/lib/PathPlanner.h:107]: This looks like a random semicolon

**Review by mjansen4857 — CHANGES_REQUESTED** [review]

**Comment by auscompgeek** [comment]
Could we get another release with this change in? Currently looking at updating the RobotPy bindings for 2023 and noticed the `generatePath` method is unusable from our end.

**Comment by virtuald** [comment]
To explain (and keep in mind for the future): RobotPy cannot use std::initializer_list since the list has to be known at compile time, so anything that uses that needs a different overload that takes a span or a vector or whatever otherwise RobotPy cannot use it. Thanks!

**Comment by mjansen4857** [comment]
Sure thing. I will look through everything to make sure there are initializer list alternatives and then push a PPLib update at least.

**Comment by mjansen4857** [comment]
OK update is now available. If there are any other issues with RobotPy support that need changes feel free to open an issue and I'll take a look.

**Pull request merged**